protocols/raft/src/main/java/io/atomix/protocols/raft/storage/snapshot/impl/DefaultSnapshotChunkReader.java

```diff
@@ -0,0 +1,56 @@
+package io.atomix.protocols.raft.storage.snapshot.impl;
+
+import io.atomix.protocols.raft.storage.snapshot.SnapshotChunk;
+import io.atomix.protocols.raft.storage.snapshot.SnapshotChunkReader;
+import io.atomix.protocols.raft.storage.snapshot.SnapshotReader;
+import java.nio.ByteBuffer;
+import java.util.NoSuchElementException;
+
+public class DefaultSnapshotChunkReader implements SnapshotChunkReader {
+  private static final int MAX_CHUNK_SIZE = 32 * 1024;
+
+  private final SnapshotReader reader;
+
+  public DefaultSnapshotChunkReader(final SnapshotReader reader) {
+    this.reader = reader;
+  }
+
+  @Override
+  public void seek(final ByteBuffer id) {
+    if (id != null) {
+      reader.skip(id.getInt(0));
+    }
+  }
+
+  @Override
+  public ByteBuffer nextId() {
+    if (hasNext()) {
+      return ByteBuffer.allocate(Integer.BYTES).putInt(0, reader.position());
+    }
+
+    return null;
+  }
+
+  @Override
+  public void close() {
+    reader.close();
+  }
+
+  @Override
+  public boolean hasNext() {
+    return reader.hasRemaining();
+  }
+
+  @Override
+  public SnapshotChunk next() {
+    if (!hasNext()) {
+      throw new NoSuchElementException();
+    }
+
+    final int offset = reader.position();
+    final ByteBuffer data = ByteBuffer.allocate(Math.min(reader.remaining(), MAX_CHUNK_SIZE));
```

Comment:
It feels not ideal/good that we have in the `next` and `nextId` an ByteBuffer.allocate. Could we have a buffer which is allocated once and you return just a view of the current chunk as byte buffer. With DirectBuffer this is easy possible idk how it is doable with the native ByteBuffers.

Reply:
That was just migrated from the previous use in AbstractAppender, I'll change it again :+1: